contracts/button/lib.rs

```diff
@@ -295,6 +300,81 @@ mod button_game {
             contract
         }
 
+        fn reset_state(&mut self) {
+            let now = self.env().block_number();
+
+            self.presses = 0;
+            self.last_presser = None;
+            self.last_press = now;
+            self.total_rewards = 0;
+
+            Self::emit_event(self.env(), Event::GameReset(GameReset { when: now }));
+        }
+
+        fn reward_pressiah(&self) -> ButtonResult<()> {
+            if let Some(pressiah) = self.last_presser {
+                let reward = self.pressiah_score();
+                self.mint_reward(pressiah, reward)??;
+            };
+
+            Ok(())
```

Comment:
I see it was like that in the previous implementation but here is philosophical question: why fn reward_pressiah ends in Ok when there is no pressiah?

Reply:
I'd say it's still `Ok(_)` in that case, because that's the expected outcome - no pressiah, no reward. An unexpected outcome would be something like a call failure due to another contract not being on chain or whatever.